src/auth/src/credentials/mds.rs

```diff
@@ -359,6 +434,51 @@ mod tests {
 
     type TestResult = anyhow::Result<()>;
 
+    #[test_case(
+        true,
+        &[
+            "retry_policy: Some(AuthRetryPolicy", 
+            "max_attempts: 5", 
+            "backoff_policy: TestBackoffPolicy", 
+            "retry_throttler: Mutex { data: AdaptiveThrottler", 
+            "factor: 4.0"
+        ];
+        "with_custom_values"
+    )]
+    #[test_case(
+        false,
+        &[
+            "retry_policy: None", 
+            "backoff_policy: ExponentialBackoff", 
+            "retry_throttler: Mutex { data: AdaptiveThrottler", 
+            "factor: 2.0"
+        ];
+        "with_default_values"
+    )]
+    fn test_mds_retry_policies(use_custom_config: bool, expected_substrings: &[&str]) {
+        let mut builder = Builder::default();
+
+        if use_custom_config {
```

Comment:
Better use two tests for this rather than a true/false argument.  Or receive the builder as an argument.